api/main.py

```diff
@@ -86,7 +108,55 @@ async def shutdown() -> None:
 
 app = LazyASGI(make_app)
 
+
+async def wait_for_s3_ready() -> None:
+    logger.info("waiting for s3 bucket to exist")
+    async with create_s3_context() as client:
+        while True:
+            try:
+                await client.head_bucket(Bucket=config.aws_bucket)
+                logger.info("s3 bucket ready")
+                break
+            except (ClientError, ClientConnectorError):
+                pass
+            await asyncio.sleep(0.1)
+
+
+async def wait_for_redis_ready() -> None:
+    logger.info("waiting for redis to be ready")
+    while True:
+        try:
+            redis = await create_redis_pool(
+                config.redis_address, config.redis_ssl_validation
+            )
+            logger.info("redis is ready")
+            redis.close()
+            await redis.wait_closed()
+            break
+        except gaierror:
+            pass
+        await asyncio.sleep(0.2)
```

Comment:
To match the other one
```suggestion
        await asyncio.sleep(0.1)
```